src/modal.js

```diff
@@ -125,8 +125,10 @@ export default class extends Controller {
     // Add classes to body to fix its position
     document.body.classList.add('fixed', 'inset-x-0', 'overflow-hidden');
 
-    // Add negative top position in order for body to stay in place
-    document.body.style.top = `-${this.scrollPosition}px`;
+    if(this.restoreScrollValue) {
```

Comment:
Should we also move `this.saveScrollPosition();` inside this conditional? If we're not restoring the scroll position, there's not much reason to save the position?

Reply:
Exactly. this is what i thought as well